test/state/widgets/actions.test.js

```diff
@@ -27,47 +28,72 @@ const STATE_MOCKED = {
+getWidgets.mockReturnValue(new Promise(resolve => resolve(PROMISE_WIDGET_LIST)));
+
+const INITIALIZE = '@@redux-form/INITIALIZE';
+
 describe('state/widgets/actions', () => {
+  let store;
+  beforeEach(() => {
+    store = mockStore(STATE_MOCKED);
+  });
+
+
   it('test SET_STATE_MOCKED for empty object on initial state', () => {
     expect(getWidgetList([])).toEqual(STATE_MOCKED);
   });
 
   it('checks action type', () => {
     const action = getWidgetList();
     expect(action.type).toBe(SET_WIDGET_LIST);
+    expect(action.type).toBe(SET_WIDGET_LIST);
   });
+
   it('search for the payload to be defined', () => {
     const action = getWidgetList();
     expect(action.payload).toBeDefined();
   });
 
-  it('action payload contains widget information', (done) => {
-    const store = mockStore(FORM_MOCK_INITIAL_STATE);
-    store.dispatch(fetchWidget(WIDGET_CODE)).then(() => {
-      const actions = store.getActions();
-      expect(actions[0].payload).toEqual(WIDGET_MOCK);
-      done();
+  describe('fetchWidget', () => {
+    it('fetchWidget calls initialize action', (done) => {
+      store.dispatch(fetchWidget()).then(() => {
+        const actions = store.getActions();
+        expect(actions[0]).toHaveProperty('type', INITIALIZE);
+        expect(actions[0]).toHaveProperty('payload', BODY_OK.payload);
+        done();
+      }).catch(done.fail);
     });
   });
 
-  it('fetchWidgetList calls setWidget action', (done) => {
-    const store = mockStore(STATE_MOCKED);
-    store.dispatch(fetchWidgetList()).then(() => {
-      const actions = store.getActions();
-      expect(actions).toHaveLength(3);
-      expect(actions[0].type).toEqual(TOGGLE_LOADING);
-      expect(actions[1].type).toEqual(SET_WIDGET_LIST);
-      expect(actions[2].type).toEqual(TOGGLE_LOADING);
-      done();
+  describe('fetchWidgetList', () => {
+    it('fetchWidgetList calls setWidgetList and setPage action', (done) => {
+      store.dispatch(fetchWidgetList()).then(() => {
+        const actions = store.getActions();
+        expect(actions).toHaveLength(4);
+        expect(actions[0]).toHaveProperty('type', TOGGLE_LOADING);
+        expect(actions[1]).toHaveProperty('type', SET_WIDGET_LIST);
+        expect(actions[1]).toHaveProperty('payload');
+        expect(actions[2].type).toEqual(TOGGLE_LOADING);
+        expect(actions[3].type).toEqual(SET_PAGE);
+        done();
+      }).catch(done.fail);
     });
-  });
 
-  it('fetchWidgetList is defined and properly valued', (done) => {
-    const store = mockStore(STATE_MOCKED);
-    store.dispatch(fetchWidgetList()).then(() => {
-      const actions = store.getActions();
-      expect(actions[1].payload.widgetList).toBeDefined();
-      done();
+    it('fetchWidgetList is defined and properly valued', (done) => {
+      store.dispatch(fetchWidgetList()).then(() => {
+        expect(getWidgets).toHaveBeenCalled();
+        const actions = store.getActions();
+        const actionPayload = actions[1].payload;
```

Comment:
Please use toHaveLenght before accessing the array elements, if the test fails the message would be more explicit